lib/tedious.js

```diff
@@ -211,13 +211,32 @@ class ConnectionPool extends base.ConnectionPool {
 
       const tedious = new tds.Connection(cfg)
 
+      // prevent calling resolve again on end event
+      let alreadyResolved = false
+      function safeResolve (err) {
+        if (!alreadyResolved) {
+          alreadyResolved = true
+          resolve(err)
+        }
+      }
+
+      function safeReject (err) {
+        if (!alreadyResolved) {
+          alreadyResolved = true
+          reject(err)
+        }
+      }
+
```

Comment:
This is the actual fix, that we are rejecting also if end is emitted, even that `connect`never triggered.

Reply:
Maybe explaining what is happening and why in a comment in the code would make sense?

Reply:
Yeah, better to add some information to comments too 👍